lib/rspec/matchers.rb

```diff
@@ -915,6 +915,7 @@ def self.configuration
 
     BE_PREDICATE_REGEX = /^(be_(?:an?_)?)(.*)/
     HAS_REGEX = /^(?:have_)(.*)/
+    BE_OR_HAS_PREDICATE_REGEX = /^(?:be_(?:an?_)?|have_).*/
```

Comment:
This could be simplified to `Regexp.union(BE_PREDICATE_REGEX, HAS_REGEX)`, which has the benefit of always being in sync with `BE_PREDICATE_REGEX` and `HAS_REGEX` when they change.

Also, I think `DYNAMIC_MATCHER_REGEX` is a better name for it, since in the context of `respond_to?` we really don't care which matchers will handle the message, just that one of the dynamic matchers will.